src/include/mutex.h

```diff
@@ -22,6 +24,27 @@ pthread_mutex_lock(&NAME##_lock);
 
 #define CMARK_UNLOCK(NAME) pthread_mutex_unlock(&NAME##_lock);
 
+#else // building for windows
+
+#define _WIN32_WINNT 0x0600
+#include <synchapi.h>
+
+#define CMARK_DEFINE_ONCE(NAME) static INIT_ONCE NAME##_once = INIT_ONCE_STATIC_INIT;
+
+#define CMARK_RUN_ONCE(NAME, FUNC) \
+BOOL pending; BOOL status; \
+status = InitOnceBeginInitialize(&NAME##_once, 0, &pending, NULL); \
+if (status && pending) { FUNC(); } \
+InitOnceComplete(&NAME##_once, 0, NULL);
```

Comment:
Actually, I think that your approach might be better, but we should wrap that in a `do { } while (0);`.  Additionally, I don't think that `InitOnceComplete` should be called on the failure.

```suggestion
#define CMARK_RUN_ONCE(name, function) do {                                     \
      BOOL fStatus;                                                             \
      BOOL fPending;                                                            \
      fStatus = InitOnceBeginInitialize(&name##_once, INIT_ONCE_ASYNC,          \
                                        &fPending, NULL);                       \
      if (!fStatus) break;                                                      \
      if (!fPending) break;                                                     \
      function();                                                               \
      (void)InitOnceComplete(&name##_once, INIT_ONCE_ASYNC, NULL);              \
    } while (0);
```

Reply:
This looks nicer - the reason i went with `InitOnceBeginInitialize` was because i'm expecting to call a function with no parameters, so i'd need to define a wrapper function to use `InitOnceExecuteOnce` properly. I'll use this version, thanks!

Reply:
Thanks for the context around the decision.  It was a bit unclear and needed a bit of thought to figure out.